pytorch_lightning/strategies/ddp.py

```diff
@@ -296,7 +296,7 @@ def _register_ddp_hooks(self) -> None:
         # In 1.8, DDP communication hooks only work with NCCL backend and SPSD (single process single device) mode
         # Since 1.9, DDP communication hooks can work on all backends.
         if _TORCH_GREATER_EQUAL_1_9 or (
-            _TORCH_GREATER_EQUAL_1_8 and self.on_gpu and self._is_single_process_single_device
+            _TORCH_GREATER_EQUAL_1_8 and self.root_device.type == "cuda" and self._is_single_process_single_device
```

Comment:
why did you choose to check it this way? in some places it may be necessary to differentiate between the model already being on gpu or not yet, but here and in most cases it is only about which accelerator is selected. So an isinstance check would have been more appropriate I believe.